### ISSUE TITLE
bug: using `python-version: 3.9-dev` installs `3.10.4`

### ISSUE THREAD

**Issue by mayeut** [actions/setup-python#416]
**Description:**

using `python-version: 3.9-dev` installs `3.10.4`

```
Run ./
  with:
    python-version: 3.9-dev
    token: ***
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.7.13/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.7.13/x64/lib/pkgconfig
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.13/x64/lib
Successfully set up CPython (3.10.4)
```

**Action version:**
v3, main

**Platform:**
- [x] Ubuntu
- [x] macOS
- [x] Windows

**Runner type:**
- [x] Hosted
- [ ] Self-hosted

**Tools version:**
`3.9-dev`

**Repro steps:**  
c.f description

**Expected behavior:**
Set up the most recent CPython 3.9 version available including pre-releases i.e. in [node-semver](https://github.com/npm/node-semver), `~3.9.0-0, {includePrerelease: true}`

**Actual behavior:**
Set up CPython 3.10.4

**Comment by panticmilos** [comment]
Hi @mayeut,

I was able to reproduce this issue, we will take look at it and related PRs today.

**Comment by dmitry-shibanov** [comment]
Hello @mayeut. The new major version was released with your fix. Could you please confirm that everything works expected?

**Comment by mayeut** [comment]
@dmitry-shibanov,
The major issue is fixed (the one from the title).
I wouldn't say "as expected": it wouldn't have installed 3.9.2rc1 back in time, it would have used a 3.9.0 or 3.9.1 instead.
However given the following, I will let you decide if you want to rename the issue or just close it:
- there are very few pre-releases from cpython x.y once a x.y.0 has been released (the last one was [3.9.2rc1](https://github.com/python/cpython/releases/tag/v3.9.2rc1) on 16 Feb 2021)
- would require #406 (but switched on automatically by the `-dev` suffix)
- [an issue in the actions/tool-cache package](https://github.com/actions/toolkit/issues/709) (In the end, I'm not sure if this one is relevant if #406 is there)

**Comment by dmitry-shibanov** [comment]
Hello @mayeut. It looks like listed pull requests were merged. Now it works as expected. For now I'm going to close the issue if you have any concerns feel free to ping us.
